fix: LEAP-1983: [FE] Conditional View Visibility Fails When Based on a Choice Selected Within a Taxonomy Tag

nass600 opened the pull request:
Restored conditional visibility working with Taxonomies via `visibleWhen` attribute. When a choice value or alias within a Taxonomy is selected it will trigger visibility in an Element that specifies it

<img width="1295" alt="Screenshot 2025-04-15 at 10 03 45" src="https://github.com/user-attachments/assets/3956086c-ac40-4b11-bcca-a997d1c6b542" />


<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nass600 committed:
fix: LEAP-1983: [FE] Conditional View Visibility Fails When Based on a Choice Selected Within a Taxonomy Tag

hlomzik commented on web/libs/editor/src/mixins/SelectedChoiceMixin.js:
```diff
@@ -40,6 +39,12 @@ const SelectedChoiceMixin = types.model().views((self) => ({
       // as they are using alias lookups for choices. For now we will keep it as is since it works for all the
       // other cases currently.
       if (self.findLabel) {
```
This `findLabel` is not used in this new code. Better to move new code upper before this check, this will be a better fit.

hlomzik commented on web/libs/editor/src/mixins/SelectedChoiceMixin.js:
```diff
@@ -40,6 +39,12 @@ const SelectedChoiceMixin = types.model().views((self) => ({
+        // For Taxonomy, we need to check if any of the selected paths contain any of the choice values
+        if (self.type === "taxonomy") {
+          return selectedValues.some(path =>
+            Array.isArray(path) && choiceValue.some(value => path.includes(value))
```
I think `path` will always be an array for taxonomies, so this check is excess, right? And if there are cases with non-array values, then we might need to handle them as well?

hlomzik replied:
Yeah, Taxonomy will always produce array values, so we don't need this check

nass600 replied:
Good improvement, thanks

nass600 committed:
code bumped up in scope

nass600 committed:
restored regular behaviour and alias

nass600 commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14486237064)
> Successfully merged:  create mode 100644 web/libs/ui/src/utils/utils.ts

nass600 committed:
Merge branch 'develop' into 'fb-leap-1983/conditional-taxonomy'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14486237064

nass600 commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14497867232)
> Successfully merged:  15 files changed, 26 insertions(+), 216 deletions(-)

nass600 committed:
Merge branch 'develop' into 'fb-leap-1983/conditional-taxonomy'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14497867232

nass600 commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14635829496)
> Successfully merged:  delete mode 100644 web/libs/ui/src/shad/components/ui/select.tsx

nass600 committed:
Merge branch 'develop' into 'fb-leap-1983/conditional-taxonomy'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14635829496

nass600 committed:
amending comparison

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14658211587)
> Successfully merged:  12 files changed, 147 insertions(+), 54 deletions(-)

MihajloHoma committed:
Merge branch 'develop' into 'fb-leap-1983/conditional-taxonomy'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14658211587

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14658253364)

MihajloHoma committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14658253364
